### PR TITLE
Try to fix CI tests

### PR DESCRIPTION
**Pull request by spxiwh**
Three changes. Two of which should reduce the frequency of the SegFault failures we've encountered a lot recently, one seems to make the CI tests avoid a segfault:

* To make the CI tests avoid a segfault, I ensure that libgomp is not first opened with RTLD_DEEPBIND. This seems to cause problems (in some environments). I don't know why. This seems to fix it.
* MKL becomes the preferred FFT backend over FFTW. MKL-breaking-FFTW segfaults won't occur if we aren't using FFTW.
* pthreads becomes the default FFTW threaded backend over openmp. openmp-related segfaults can't occur if we are using pthreads for threaded FFTW.

This then exposed a few extra things:

* The LISA example doesn't work with MKL because of https://github.com/gwastro/pycbc/issues/4395
* The MKL error catching code didn't work and led to a segfault. This is fixed so we catch and print the error correctly.

### CODE REVIEW

**Comment by ahnitz** [comment]
@spxiwh Can we add an error message to let the user know what to do if they do try to use a > 2**24 point FFT with MKL? I think it might be worth it in this case.

**Comment by spxiwh** [comment]
@ahnitz Yes, it would be good to add this.

Can we do this separately from this PR though? This PR doesn't change any behaviour in MKL and I'd prefer to limit this to "let's just get the CI suite working again" and then consider MKL and how to handle longer arrays separately (I'm hoping there is an easy way to get MKL to work on longer arrays. The "descriptor" object seems to be the place to look, but the documentation is ... not great.)

**Review by ahnitz — APPROVED** [review]

**Comment by josh-willis** [comment]
So for some reason when I install from this branch and run a basic threading test I had setup, setting the threading backend to `openmp` isn't actually changing the threading backend (e.g., using ` --fftw-threads-backend openmp` in a program using our builtin CLI functions). This could be some problem with my test as I don't see how the changes on this branch could have led to that, but I want to understand what's going on first...

**Comment by josh-willis** [comment]
> So for some reason when I install from this branch and run a basic threading test I had setup, setting the threading backend to `openmp` isn't actually changing the threading backend (e.g., using ` --fftw-threads-backend openmp` in a program using our builtin CLI functions). This could be some problem with my test as I don't see how the changes on this branch could have led to that, but I want to understand what's going on first...

Indeed, the problem was with my test, which specified `--fftw-threads-backend openmp` but not `--fft-backend`. With this patch of course I must do so to exercise FFTW as the backend.

For me, this patch also seems to resolve the segfault error in a conda environment which has  all of MKL, BLAS linked to MKL, and liblal linked to FFTW. Not sure why it would do that, but that's great, since it's much simpler than what I had been working on.  I will add a review with a couple of minor requests but this has my in-principle approval.

File: pycbc/fft/fftw.py
Comment (josh-willis) [inline pycbc/fft/fftw.py:111]: Can this test instead be:
```
if backend == 'openmp':
```
since I think it's clearer?

File: pycbc/fft/fftw.py
Comment (josh-willis) [inline pycbc/fft/fftw.py:160]: Please also change the comment line immediately above to indicate that now 'pthreads' is the first thing tried, or we'll get very confused later.

**Review by josh-willis — CHANGES_REQUESTED** [review]
I've requested changes but since Alex has approved, once the changes are made (at least the one about updating the comment line please) consider this approved from me and go ahead and merge, so CI is back to working.

**Review by josh-willis — APPROVED** [review]

**Comment by josh-willis** [comment]
I would still like the changes but am removing "Changes requested" so that this does not have to wait on me approving said changes...

**Comment by spxiwh** [comment]
Thanks Josh. The changes suggested make sense. I'll implement this and then get this merged.

**Pull request merged**
